File: e2e/cypress-javascript/interactions/pages/TaskInfoModal.js
class TaskInfoModal {
  get title() {
    return cy.get('[data-testid="modal-title"]');
  }

  get description() {
    return cy.get('[data-testid="description"]');
  }

  get validationBadge() {
    return cy.get('[data-testid="valid"]');
  }

  statusTag(status) {
    return cy.get('[data-testid="status"]').find(`[data-testid="status-tag-${status}"]`);
  }

  priorityTag(priority) {
    return cy.get('[data-testid="priority"]').find(`[data-testid="priority-tag-${priority}"]`);
  }
}

module.exports = { TaskInfoModal };

File: e2e/capybara-ruby/interactions/pages/task_info_modal.rb
# frozen_string_literal: true

require_relative '../../support/capybara_helpers'

class TaskInfoModal
  include CapybaraHelpers
  def title
    find('[data-testid="modal-title"]')
  end

  def description
    find('[data-testid="description"]')
  end

  def validation_badge
    find('[data-testid="valid"]')
  end

  def status_tag(status)
    find('[data-testid="status"]').find("[data-testid=\"status-tag-#{status}\"]")
  end

  def priority_tag(priority)
    find('[data-testid="priority"]').find("[data-testid=\"priority-tag-#{priority}\"]")
  end
end
